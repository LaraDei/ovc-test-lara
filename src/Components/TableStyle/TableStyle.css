.table{
    width: 85%;
    margin-left: auto;
    margin-right: auto;
}

table, th, td {
    border: 1px solid black;
    border-collapse: collapse;
    padding: 8px 18px;
}

tbody tr:hover {
    background-color: #ffff99;
}
  
thead {
  background-color: lightgrey;
}





