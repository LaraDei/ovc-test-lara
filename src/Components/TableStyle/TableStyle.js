import React, {Component} from 'react';

import './TableStyle.css';

export default class TableStyle extends Component{

    renderTableHeader(){
        return this.props.columns.map((title, index)=>{
            return (
                <th key={index+title}>{title}</th>
            )
        })
    }
    renderTableBody(){
        return this.props.data.map(item => {
           return (
               <tr key={item.userId} id={item.userId} onClick={(e)=>this.props.function(e)}>
                   {Object.keys(item).map((key, index) => {
                        if(index >0){
                            return (
                            <td key={item[key]}>{item[key]}</td>
                            )
                        }else return null;
                })}
               </tr>
            )
        })
    }


    render(){
        return(
            <div className="table-container">
                <table className="table">
                    <thead>
                        <tr>
                            {this.renderTableHeader()}
                        </tr>
                    </thead>
                    <tbody>
                        {this.renderTableBody()}
                    </tbody>
                </table>
                
            </div>
        )
    }
    
}