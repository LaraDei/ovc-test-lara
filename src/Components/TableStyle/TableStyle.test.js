import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import TableStyle from "./TableStyle";

describe("TableStyle component", () => {
  const props = {
    columns: ["test1", "test2", "test3"],
    data: [
      {
        id: 123,
        name: "testName",
        email: "testEmal",
        city: "testCity",
        company: "testCompany",
      },
    ],
  };
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TableStyle {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const wrapper = renderer.create(<TableStyle {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a header for each item in given array", () => {});
  it("it renders a row for each item in given array", () => {});
  it("it executes a function when row is clicked", () => {});
});
