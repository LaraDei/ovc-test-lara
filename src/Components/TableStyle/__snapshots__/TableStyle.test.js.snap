// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableStyle component renders by default 1`] = `
<div
  className="table-container"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          test1
        </th>
        <th>
          test2
        </th>
        <th>
          test3
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        id={123}
        onClick={[Function]}
      >
        <td>
          testName
        </td>
        <td>
          testEmal
        </td>
        <td>
          testCity
        </td>
        <td>
          testCompany
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TableStyle component renders the UI as expected 1`] = `
<div
  className="table-container"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          test1
        </th>
        <th>
          test2
        </th>
        <th>
          test3
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        id={123}
        onClick={[Function]}
      >
        <td>
          testName
        </td>
        <td>
          testEmal
        </td>
        <td>
          testCity
        </td>
        <td>
          testCompany
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TableStyle component renders the UI as expected with empty props 1`] = `
<div
  className="table-container"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          test1
        </th>
        <th>
          test2
        </th>
        <th>
          test3
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        id={123}
        onClick={[Function]}
      >
        <td>
          testName
        </td>
        <td>
          testEmal
        </td>
        <td>
          testCity
        </td>
        <td>
          testCompany
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
